### PR TITLE
Make new paint code use integrated variables

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/interface/viewport.c [hunk 1]
```diff
@@ -1331,7 +1331,7 @@ bool sub_68818E(uint32 image_id, uint16 x, uint16 y, paint_struct ** paint)
 		return false;
 	}
 
-	RCT2_GLOBAL(0x00EE7888, uint32) += 0x12;
+	unk_EE7888 = (paint_struct *)((uintptr_t)unk_EE7888 + 0x12);
```
Comment (janisozaur) [inline src/interface/viewport.c]: It matches old code, but is the old code OK? It looks bogus.

**Comment by janisozaur** [comment]
This is for `port/x64`. When you apply this patch as well:
```diff
diff --git a/src/common.h b/src/common.h
index 56dd209..9c32f80 100644
--- a/src/common.h
+++ b/src/common.h
@@ -36,6 +36,7 @@
 
 #if defined(__i386__) || defined(_M_IX86)
 #define PLATFORM_X86
+#define NO_RCT2 1
 #else
 #define NO_RCT2 1
 #endif
```
it works on 32 bit again, but the results are… funky.

x86:
openrct2_117

**Comment by Joshwoo69** [comment]
Looks to me like no terrain and under ground paths visible.

File: src/interface/viewport.c (hunk 1 above)
Comment (duncanspumpkin) [inline src/interface/viewport.c]: EE7888 is a pointer to the next available paint struct. There are 3 different types of paint struct with 3 different sizes. If you have a look at https://github.com/OpenRCT2/OpenRCT2/pull/3438 we are making this all more obvious. I'm not sure what you want to do for the x64 port stuff.

**Pull request merged**